crates/turbopack-node/src/render/render_proxy.rs

```diff
@@ -172,13 +131,167 @@ async fn proxy_error(
     .as_issue()
     .emit();
 
-    Ok(ProxyResult {
-        status: status_code,
-        headers: vec![(
-            "content-type".to_string(),
-            "text/html; charset=utf-8".to_string(),
-        )],
-        body: body.clone().into(),
+    Ok((status_code, body))
+}
+
+#[derive(Clone, Debug)]
+#[turbo_tasks::value]
+enum RenderItem {
+    Headers(ResponseHeaders),
+    BodyChunk(Bytes),
+}
+
+type RenderItemResult = Result<RenderItem, SharedError>;
+
+#[turbo_tasks::value(eq = "manual", cell = "new", serialization = "none")]
+pub struct RenderStreamSender {
+    #[turbo_tasks(trace_ignore, debug_ignore)]
+    get: Box<dyn Fn() -> UnboundedSender<RenderItemResult> + Send + Sync>,
+}
+
+#[turbo_tasks::value(transparent, eq = "manual", cell = "new", serialization = "none")]
+struct RenderStream(#[turbo_tasks(trace_ignore)] Stream<RenderItemResult>);
+
+#[turbo_tasks::function]
+fn render_stream(
+    pool: NodeJsPoolVc,
+    data: RenderDataVc,
+    body: BodyVc,
+    intermediate_asset: AssetVc,
+    intermediate_output_path: FileSystemPathVc,
+    project_dir: FileSystemPathVc,
+    error_path: FileSystemPathVc,
+) -> RenderStreamVc {
+    // Note the following code uses some hacks to create a child task that produces
+    // a stream that is returned by this task.
+
+    // We create a new cell in this task, which will be updated from the
+    // [render_stream_internal] task.
+    let cell = turbo_tasks::macro_helpers::find_cell_by_type(*RENDERSTREAM_VALUE_TYPE_ID);
```

Comment:
We probably should move these hacks into turbo-tasks-bytes

Reply:
I actually think we should move `Stream<T>` into `turbo-tasks` proper, and design a macro interface as you suggested.